src/parsers/minecraft/nbt/doc_walker.ts

```diff
@@ -0,0 +1,168 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as url from "url";
+import { runNodeFunction } from "./doc_walker_func";
+import {
+    CompoundNode,
+    FunctionNode,
+    isCompoundNode,
+    isFunctionNode,
+    isListNode,
+    isRefNode,
+    isRootNode,
+    NBTNode,
+    RefNode
+} from "./doc_walker_util";
+import { NBTTag } from "./tag/nbt_tag";
+import { ArrayReader } from "./util/array_reader";
+
+export type ValueList = string[];
+
+const rootNodePath = require.resolve("mc-nbt-paths/root.json");
+
+export class NBTWalker {
+    private readonly parsed: NBTTag<any>;
+    private readonly root: string;
+
+    public constructor(parsed: NBTTag<any>, root: string = rootNodePath) {
+        this.parsed = parsed;
+        this.root = root;
+    }
+
+    public getFinalNode(nbtpath: string[]): NBTNode | undefined {
+        const rootNode = JSON.parse(
+            fs.readFileSync(this.root).toString()
```

Comment:
We should be doing all reading into memory at startup time. It could be as simple as creating a basic filesystem proxy object.

This blocks loading nbt on I/O operations, which are unchecked so may fail. This is especially problematic for suggestions, as the speed of suggestions is important - we don't want uneccessary blocking IO.